### PR TITLE
Context task

### PR DESCRIPTION
**Pull request by DDianovMD**

### CODE REVIEW

File: src/queries/employeeQueries.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { useQuery, useMutation } from "react-query";
+import {
+  getEmployees,
+  getEmployeeById,
+  addEmployee,
+  updateEmployee,
+  deleteEmployee,
+} from "../api/apicalls";
+
+export const useAddEmployeeMutation = (onSuccess: (data: any) => void, onError: (error: any) => void) => {
```
Comment (mariyan-marinov) [inline src/queries/employeeQueries.ts]: no need of custom hooks, remove the 'use'

File: src/toast/messages.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { Toast } from "primereact/toast";
+
+export const showAdded = (toast: React.RefObject<Toast>): void => {
```
Comment (mariyan-marinov) [inline src/toast/messages.tsx]: too much same code..
create method calls
 toast.current?.show({
    severity: "...",
    summary: "Info",
    detail: "...",
    life: 3000,
  }); and reuse it in all of these methods.

Also best approach would be to have showInfo, showWarning, showSuccess like methods that just accept detail string as param. Would be much easier to call them, more readable, etc.

**Review by mariyan-marinov — CHANGES_REQUESTED** [review]

File: src/Components/prime-react-homepage/PrimeReactHomePage.tsx [hunk 3]
```diff
@@ -156,17 +153,16 @@ export function PrimeReactHomePage() {
...
             setEditVisibility(false);
           }}
         >
-          <Edit
-            toast={toast}
-            setEditVisibility={setEditVisibility}
-            employee={employee}
-          />
+          <EmployeeContext.Provider value={employee}>
+            <Edit toast={toast} setEditVisibility={setEditVisibility} />
```
Comment (mariyan-marinov) [inline src/Components/prime-react-homepage/PrimeReactHomePage.tsx]: setEditVisibility is not used anymore .. and this is the right approach. Visibility should be handled outside of the component

File: src/Components/edit/Edit.tsx [hunk 4]
```diff
@@ -26,7 +29,9 @@ const Edit = ({ toast, setEditVisibility, employee }: EditProps) => {
       if (employee !== editFormik.initialValues) {
         updateMutation.mutate(employee);
       } else {
-        showInfoMessage(toast);
+        const messageSummary = "No changes were made.";
+        const message = "In order to update information you should change field values.";
+        showMessage(toast, "info", messageSummary, message);
```
Comment (mariyan-marinov) [inline src/Components/edit/Edit.tsx]: do not use showMessage, consider it as private method. create method like showInfo(messageSummary, message), showWarning .. Inside showInfo call showMessage with corresponding params

**Review by mariyan-marinov — CHANGES_REQUESTED** [review]

File: src/Models/EmployeesContext.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { createContext } from 'react';
+import { Employee } from './Employee';
+
+export const EmployeesContext = createContext<Employee[] | undefined>(undefined);
```
Comment (mariyan-marinov) [inline src/Models/EmployeesContext.ts]: what is the purpose of this?

**Review by mariyan-marinov — CHANGES_REQUESTED** [review]

File: src/Components/prime-react-homepage/PrimeReactHomePage.tsx (hunk 3 above)
Comment (DDianovMD) [reply src/Components/prime-react-homepage/PrimeReactHomePage.tsx]: It's used to set sidebar's "visible" prop to "false" resulting in hiding sidebar when submitting the form with new values.

File: src/queries/employeeQueries.ts (hunk 1 above)
Comment (DDianovMD) [reply src/queries/employeeQueries.ts]: Fixed.

File: src/Components/edit/Edit.tsx (hunk 4 above)
Comment (DDianovMD) [reply src/Components/edit/Edit.tsx]: Fixed.

File: src/toast/messages.tsx (hunk 2 above)
Comment (DDianovMD) [reply src/toast/messages.tsx]: Done.

File: src/Components/prime-react-homepage/PrimeReactHomePage.tsx (hunk 3 above)
Comment (mariyan-marinov) [reply src/Components/prime-react-homepage/PrimeReactHomePage.tsx]: but is not used in the Edit component, remove it from there

File: src/Models/EmployeesContext.ts (hunk 5 above)
Comment (DDianovMD) [reply src/Models/EmployeesContext.ts]: Deleted.

**Review by mariyan-marinov — APPROVED** [review]

**Pull request merged**